app/controllers/users_controller.rb

```diff
@@ -2,6 +2,7 @@ class UsersController < ApplicationController
   before_action :set_user, only: :show
 
   def show
+    @comments = Assignment.where(submitter_id: @user.id).try(:flat_map, &:comments) || []
```

Comment:
Here again you are repeating the join logic. You can use the association I mentioned in my previous comments